qtaf-core/src/main/java/de/qytera/qtaf/core/log/model/index/LogMessageIndex.java

```diff
@@ -122,4 +123,55 @@ public List<LogMessage> getByScenarioId(String scenarioId) {
                 .filter(m -> m.getScenarioId().equals(scenarioId))
                 .collect(Collectors.toList());
     }
+
+    /**
+     * Get all log messages for a given scenario ID that are pending
+     *
+     * @param scenarioId Scenario ID
+     * @return List of log messages for the given scenario ID
+     */
+    public List<LogMessage> getByScenarioIdAndPending(String scenarioId) {
+        return this.index
+                .values()
+                .stream()
+                .filter(m -> m instanceof StepInformationLogMessage
+                        && m.getScenarioId().equals(scenarioId)
+                        && ((StepInformationLogMessage) m).getStatus() == StepInformationLogMessage.Status.PENDING
+                )
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Get all log messages for a given scenario ID that have passed
+     *
+     * @param scenarioId Scenario ID
+     * @return List of log messages for the given scenario ID
+     */
+    public List<LogMessage> getByScenarioIdAndPassed(String scenarioId) {
+        return this.index
+                .values()
+                .stream()
+                .filter(m -> m instanceof StepInformationLogMessage
+                        && m.getScenarioId().equals(scenarioId)
+                        && ((StepInformationLogMessage) m).getStatus() == StepInformationLogMessage.Status.PASS
+                )
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Get all log messages for a given scenario ID that have failed
+     *
+     * @param scenarioId Scenario ID
+     * @return List of log messages for the given scenario ID
+     */
+    public List<LogMessage> getByScenarioIdAndFailed(String scenarioId) {
+        return this.index
+                .values()
+                .stream()
+                .filter(m -> m instanceof StepInformationLogMessage
+                        && m.getScenarioId().equals(scenarioId)
+                        && ((StepInformationLogMessage) m).getStatus() == StepInformationLogMessage.Status.ERROR
+                )
+                .collect(Collectors.toList());
```

Comment:
This can be simplified to:

```java
   public List<StepInformationLogMessage> getByScenarioIdAndPending(String scenarioId) {
        return this.index
                .values()
                .stream()
                .filter(StepInformationLogMessage.class::isInstance)
                .map(StepInformationLogMessage.class::cast)
                .filter(step -> step.getScenarioId().equals(scenarioId) && step.getStatus() == StepInformationLogMessage.Status.ERROR)
                .toList();
    }
```

It also removes redundant `StepInformationLogMessage` casts in calling methods.